packages/components/src/Link/Link.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { PropsWithChildren } from "react";
+
+interface LinkProps {
+  readonly ariaLabel?: string;
+  readonly external?: boolean;
+  readonly url: string;
+}
+
+export function Link({
+  ariaLabel,
+  external = false,
+  url,
+  children,
+}: PropsWithChildren<LinkProps>) {
```

Comment:
It occurred to me that we might want an `onClick` event as well. Thoughts?

Reply:
Considered that.. it opens the door to a lot of anti-patterns.  The legit use case I can see might be needing to add dynamic behavior around a "you'll lose your work if you leave" type dialog.  But: that's kind of an anti-pattern to (vs e.g. using window.beforeunload where you can do such checks in one place instead of trying to intercept anything that might take you off the page).

Are there other (commendable) use cases we can see right now?  Otherwise, I'm inclined to opine in favour of YAGNI.

Reply:
Here's what autopilot had to say about it when I started typing this sentence 😂 

> Adding a prop for OnClick was considered, but rejected as it would be too easy to abuse. If you need to trigger an action, use a `<Button>` instead.

Reply:
Is this kind of the guideline that you are thinking? https://polaris.shopify.com/components/navigation/link#submitting-data

i.e. links should navigate, and from a UI perspective other things (like submitting data, or triggering other dynamic behaviour) should look like buttons (or, at least not look like links?)
I think this makes sense, curious what others think:  @fagnanm @darryltec

Reply:
Yeah - essentially.  "buttons" are the place where I would want to see flexibility (to whatever extent that Button component supports those use cases is a separate concern)..  My thinking is that links are a fairly special/consistent thing users expect to behave a certain way.

Reply:
Yes that makes sense. TBH it's never come up on the design team, may have to push on it a little more if we had a button style that also looked like a link in the future.

Reply:
Oh nice! I was going to check active usages... The real question is do we want these to be `<Link>`'s under the hood, or should they be `<Button>`'s styled to look like links?

Reply:
we have a shiny new eslint rule for a11y where it will force you to use a button because screen readers and keyboard